src/sema/expression.rs

```diff
@@ -1238,6 +1246,55 @@ impl Expression {
                     expr: Box::new(self.clone()),
                 })
             }
+            (Type::FunctionSelector, Type::Bytes(n)) => {
+                let selector_length = ns.target.selector_length() as u8;
+                if *n < selector_length {
+                    diagnostics.push(Diagnostic::cast_error(
+                        *loc,
+                        format!(
+                            "function selector can only be casted to bytes{} or larger",
+                            selector_length
+                        ),
+                    ));
+                    Err(())
+                } else if *n == selector_length {
+                    Ok(Expression::Cast {
+                        loc: *loc,
+                        to: to.clone(),
+                        expr: self.clone().into(),
+                    })
+                } else {
+                    self.cast_types(
+                        loc,
+                        &Type::Bytes(selector_length),
+                        to,
+                        implicit,
+                        ns,
+                        diagnostics,
+                    )
+                }
+            }
+            (Type::FunctionSelector, Type::Uint(n) | Type::Int(n)) => {
```

Comment:
Why do we need conversion to `uint64` and even `int64`? Seems wrong.

Reply:
`Bytes(8)` can be converted to `uint64` and `int64`.

Reply:
I changed the error I wrote there to a warning to be more consistent with the error we get when converting `bytes8` to `uint64` (implicit cast now allowed).